lib/miq_automation_engine/engine/miq_ae_method.rb

```diff
@@ -104,13 +115,21 @@ def self.invoke_external(cmd, workspace, serialize_workspace = false)
         end
         rc  = status.exitstatus
         msg = "Method exited with rc=#{verbose_rc(rc)}"
+        method_pid = nil
+        threads = []
       rescue => err
         $miq_ae_logger.error("Method exec failed because (#{err.class}:#{err.message})")
         rc = MIQ_ABORT
         msg = "Method execution failed"
+      ensure
+        if method_pid
+          $miq_ae_logger.error("Terminating non responsive method with pid #{method_pid.inspect}")
+          Process.kill("TERM", method_pid) rescue nil
```

Comment:
I'm concerned about a race condition between not clearing up the `method_pid` and the process dieing. So I want to catch a couple of the `Errno::*`. Not sure if we want to spend time trying to figure out all the ways this could fail. I was thinking `rescue nil` may just be good enough